File: MegaHerdt.Models/Models/Shipment.cs

using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MegaHerdt.Models.Models
{
    public class Shipment
    {
        [Key]
        public int Id { get; set; }     
        [ForeignKey("Address")]
        public int AddressId { get; set; }
        [ForeignKey("Purchase")]
        public int PurchaseId { get; set; }
        [ForeignKey("TransportCompany")]
        public int? TransportCompanyId { get; set; }
        public float Amount { get; set; }
        public DateTime ShipmentDate { get; set; }
        public int TrackingNumber { get; set; }
        public Purchase Purchase { get; set; }
        public Address Address { get; set; }
        public TransportCompany? TransportCompany { get; set; }

    }
}

File: MegaHerdt.FrontEnd/MegaHerdt/src/app/models/Shipment/Shipment.ts
import { AddressUpdate } from "../Address/AddressUpdate";
import { TransportCompany } from "../TransportCompany/TransportCompany";

export class Shipment{
    id: number;
    purchaseId: number;
    shipmentDate: Date;
    amount: number;
    trackingNumber: number;
    address: AddressUpdate;
    transportCompany: TransportCompany;
    constructor(){
        this.id = 0;
        this.purchaseId = 0;
        this.shipmentDate = new Date();
        this.amount = 0;
        this.trackingNumber = 0;
        this.address = new AddressUpdate(0,'',0,'',0,'','','');
        this.transportCompany = new TransportCompany();

    }
}